### PR TITLE
media-sound/nulloy add

### PR DESCRIPTION
**Pull request by vitaly-zdanevich**

### CODE REVIEW

**Review by thesamesam — COMMENTED** [review]
Ebuild looks good!

Some notes:
- We don't usually add new packages straight away, we usually put them into GURU first, or see if they're popular.
- We'd want to add a release version as well or at least a snapshot (not just a live ebuild)
- For adding new people as maintainers, we want them to have made a few PRs to existing packages first. Anything maintainer-needed you have installed with open bugs / needing a version bump, for example?
- I'd just squash all the commits based on CI results (then force push; this is fine to your own branch(es))

File: media-sound/nulloy/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	<maintainer type="person" proxied="yes">
+		<email>zdanevich.vitaly@ya.ru</email>
+		<name>Vitaly Zdanevich</name>
+	</maintainer>
+	<maintainer type="project" proxied="proxy">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
+	<maintainer type="project">
+		<email>sound@gentoo.org</email>
```
Comment (thesamesam) [inline media-sound/nulloy/metadata.xml:13]: Drop this part unless someone from @gentoo/sound said they want it

**Comment by vitaly-zdanevich** [comment]
> For adding new people as maintainers, we want them to have made a few PRs to existing packages first. Anything maintainer-needed you have installed with open bugs / needing a version bump, for example?

I have this pull request https://github.com/gentoo/gentoo/pull/21214

**Comment by fordfrog** [comment]
@vitaly-zdanevich
* you can put back sound@gentoo.org as the maintainer, i have no problem with that
* otherwise, as @thesamesam mentioned, make it all one commit and also create an ebuild for a versioned release. you can make a snapshot version in this case (something like 0.8.2_p20211228) as the last release was back in 2016 afaics

**Comment by vitaly-zdanevich** [comment]
> * the last release was back in 2016

[Asked upstream](https://github.com/nulloy/nulloy/issues/158#issuecomment-1013056150) to release current master, in order to have fresh version.

**Comment by vitaly-zdanevich** [comment]
@fordfrog 

Upstream released `0.9`, I tested it - works ok, so this is a good time to continue with this ebuild.

Will I need to publish to Guru first?

**Comment by vitaly-zdanevich** [comment]
Also I want to [add my skin - developer did not accept my PR](https://github.com/nulloy/nulloy/pull/181), what is the correct way to do it?

**Comment by fordfrog** [comment]
> @fordfrog
> 
> Upstream released [0.9](https://github.com/nulloy/nulloy/releases/tag/0.9), I tested it - works ok, so this is a good time to continue with this ebuild.
> 
> Will I need to publish to Guru first?

no need for guru. just please squash the commits into one.

wrt the skin, i suggest you add a use flag for it (you can call it `custom-skins` for example) and if enabled, install also the custom skin.

**Comment by vitaly-zdanevich** [comment]
> no need for guru. just please squash the commits into one.

Thanks! Will squash the commits when ebuilds will be ready.

About skin: do you have some good example about what I need to add to ebuild about making zip, copying to `/usr/share/nulloy/skins/night.nzs` - as I see ebuild has a lot of magic commands, or just `zip` and `cp` will be enough?

**Comment by vitaly-zdanevich** [comment]
@fordfrog I figured out how to add theme, please check.

**Comment by fordfrog** [comment]
> @fordfrog I figured out how to add theme, please check.

nice. i think it would be better though to package the theme as a tarball and download it from somewhere rather than putting it into the gentoo repo directly.

**Comment by vitaly-zdanevich** [comment]
@fordfrog created https://gitlab.com/vitaly-zdanevich/nulloy-theme-night, ebuild must somehow git clone this repo?

**Comment by vitaly-zdanevich** [comment]
> you should download a versioned tarball of the theme

Download with wget?

**Comment by thesamesam** [comment]
> Looks like `wget` is not an option - tried twice, got `Resolving gitlab.com... failed: Temporary failure in name resolution`, the same command works as expected in terminal.

You cannot make network connections within the ebuild other than via SRC_URI. You can have multiple entries (space-separated) in SRC_URI.

**Comment by vitaly-zdanevich** [comment]
Thanks, how can I add conditionally to SRC_URI, according to USE flag?

**Comment by thesamesam** [comment]
> Thanks, how can I add conditionally to SRC_URI, according to USE flag?

https://devmanual.gentoo.org/ebuild-writing/variables/#conditional-sources

**Comment by vitaly-zdanevich** [comment]
> for live version of the ebuild, you can see an example how to use multiple repos [here](https://github.com/gentoo/gentoo/blob/master/net-wireless/iwd/iwd-9999.ebuild#L122-L123).

Did like that, faced with a problem that after my `yes` this process is frozen:

```
>>> Emerging (1 of 1) media-sound/nulloy-9999::gentoo
>>> Unpacking source...
 * Repository id: git@gitlab.com:vitaly-zdanevich_nulloy-theme-night.git
 * To override fetched repository properties, use:
 *   EGIT_OVERRIDE_REPO_GIT_GITLAB_COM_VITALY_ZDANEVICH_NULLOY_THEME_NIGHT
 *   EGIT_OVERRIDE_BRANCH_GIT_GITLAB_COM_VITALY_ZDANEVICH_NULLOY_THEME_NIGHT
 *   EGIT_OVERRIDE_COMMIT_GIT_GITLAB_COM_VITALY_ZDANEVICH_NULLOY_THEME_NIGHT
 *   EGIT_OVERRIDE_COMMIT_DATE_GIT_GITLAB_COM_VITALY_ZDANEVICH_NULLOY_THEME_NIGHT
 * 
 * Fetching git@gitlab.com:vitaly-zdanevich/nulloy-theme-night.git ...
git fetch git@gitlab.com:vitaly-zdanevich/nulloy-theme-night.git +HEAD:refs/git-r3/HEAD --depth 1
The authenticity of host 'gitlab.com (172.65.251.78)' can't be established.
ED25519 key fingerprint is SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
```

Usual git clone from this repo is normal.

**Comment by vitaly-zdanevich** [comment]
Switching from ssh to http helped.

**Comment by vitaly-zdanevich** [comment]
About 9999: `eapply` stopped to work, now I see:

```
 * Applying nulloy-9999.patch ...
can't find file to patch at input line 4
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|index 4ddee63..5f02c60 100644
|--- a/src/skins/skins.pri
|+++ b/src/skins/skins.pri
--------------------------
No file to patch.  Skipping patch.
```

**Comment by vitaly-zdanevich** [comment]
> You can have multiple entries (space-separated) in SRC_URI.

Looks like the last EGIT_REPO_URI overwrites the first one. What is the correct way to clone both repos for 9999?

**Comment by fordfrog** [comment]
could you first merge all commits into one please? and you should also fix the qa warnings first.

**Comment by vitaly-zdanevich** [comment]
> could you first merge all commits into one please?

I am not experienced in this, what is the best way? Maybe some button in Github? `git merge --squash` with new PR?

**Comment by vitaly-zdanevich** [comment]
Something is wrong with my rebase, I will make another PR.

**Pull request closed without merging**